app/models/network_port.rb

```diff
@@ -0,0 +1,17 @@
+class NetworkPort < ActiveRecord::Base
+  include NewWithTypeStiMixin
+  include ReportableMixin
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::CloudManager"
+  belongs_to :cloud_tenant
+  belongs_to :cloud_network
+  belongs_to :cloud_subnets
```

Comment:
Yeah that is the thing, the cloudy stuff is really for Neutron. An OpenstackInfra also has Neutron, in this case the device of the port will be infra Host.